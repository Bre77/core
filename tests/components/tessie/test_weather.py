"""Test the Tessie weather platform."""
from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant

from .common import TEST_WEATHER, setup_platform


async def test_weather(
    hass: HomeAssistant, mock_get_weather, snapshot: SnapshotAssertion
) -> None:
    """Tests that the weather entity is correct."""

    assert len(hass.states.async_all("weather")) == 0

    await setup_platform(hass)

    assert len(hass.states.async_all("weather")) == 1

    mock_get_weather.assert_called_once()
    state = hass.states.get("weather.test")
    assert state.state == TEST_WEATHER["condition"]
    assert state == snapshot
