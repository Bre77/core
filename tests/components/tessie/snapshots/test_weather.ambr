# serializer version: 1
# name: test_weather
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'apparent_temperature': 23.5,
      'cloud_coverage': 100,
      'friendly_name': 'Test',
      'humidity': 83,
      'precipitation_unit': <UnitOfPrecipitationDepth.MILLIMETERS: 'mm'>,
      'pressure': 1010.0,
      'pressure_unit': <UnitOfPressure.HPA: 'hPa'>,
      'temperature': 23.0,
      'temperature_unit': <UnitOfTemperature.CELSIUS: 'Â°C'>,
      'visibility': 10.0,
      'visibility_unit': <UnitOfLength.KILOMETERS: 'km'>,
      'wind_bearing': 165,
      'wind_speed': 0.45,
      'wind_speed_unit': <UnitOfSpeed.KILOMETERS_PER_HOUR: 'km/h'>,
    }),
    'context': <ANY>,
    'entity_id': 'weather.test',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'Rain',
  })
# ---
